example-permit/sources/main.move

```diff
@@ -0,0 +1,164 @@
+module example_permit::main {
+    use aptos_framework::object;
+    use minter::token_minter;
+    use std::ed25519;
+    use std::option;
+    use std::signer::{address_of};
+    use std::string::{String, utf8};
+
+    /// The provided permit is bad.
+    const EBAD_PERMIT: u64 = 1;
+
+    struct AppConfig has key {
+        authority_public_key: ed25519::UnvalidatedPublicKey,
+        extend_ref: object::ExtendRef,
+        token_minter_obj: object::Object<token_minter::TokenMinter>,
+    }
+
+    struct MintPermit has copy, drop {
+        name: String,
+        description: String,
+        uri: String,
+        recipient_addr: address,
+    }
+
+    /// Utility to update the public key. NOTE: This is unauthenticated and should only be used for testing.
+    public entry fun set_public_key(authority_public_key: vector<u8>) acquires AppConfig {
+        let app_config = borrow_global_mut<AppConfig>(@example_permit);
+        app_config.authority_public_key = ed25519:: new_unvalidated_public_key_from_bytes(authority_public_key);
+    }
+
+    public entry fun init(deployer: &signer, authority_public_key: vector<u8>) {
+        let constructor_ref = &object::create_object(address_of(deployer));
+        let token_minter_obj = token_minter::init_token_minter_object(
+            &object::generate_signer(constructor_ref),
+            utf8(b"Permit Collection"),
+            option::none(),
+            utf8(b"Collection for permit mints"),
+            utf8(b"https://example.com/example_permit"),
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            option::none(),
+            true, // creator_mint_only
+            false,
+        );
+        move_to(
+            deployer,
```

Comment:
for my own sake, at what point we decide who should own what? like in this case, `AppConfig` is owned by `deployer`, but an object owned by `deployer` actually owns `token_minter_object`